base/docs/basedocs.jl

```diff
@@ -2103,4 +2098,36 @@ Union type of [`StridedVector`](@ref) and [`StridedMatrix`](@ref) with elements
 """
 StridedVecOrMat
 
+"""
+    Module
+
+A `Module` is a separate global variable workspace. See [`module`](@ref) and the [manual section about modules](@ref modules) for details.
+"""
+Module
+
+"""
+    Core
+
+`Core` is the module that contains all identifiers considered "built in" to the language, i.e. part of the core language and not libraries. Every module implicitly specifies `using Core`, since you can't do anything without those definitions.
+"""
+kw"Core"
+
+"""
+    Main
+
+`Main` is the top-level module, and Julia starts with `Main` set as the current module.  Variables defined at the prompt go in `Main`, and `varinfo` lists variables in `Main`.
+```jldoctest
+julia> @__MODULE__
+Main
+```
+"""
+kw"Main"
+
+"""
+    `Base`
+
+The base library of Julia. `Base` is a module that contains basic functionality (the contents of `base/`). All modules implicitly contain `using Base`, since this is needed in the vast majority of cases.
+"""
+kw"Base"
```

Comment:
I'm not sure why these would be handled like keywords. Can we attach the doc strings to the modules themselves instead?

Reply:
Note that `kw"Base"` was introduced before this PR. I guess maybe for some kind of bootstrap reason? Anyway in case of `Main` and `Core` I needed to do this in order to make e.g.
```julia
help?> Core
```
work.

Reply:
This is what I get, if I remove `kw`:
```julia
help?> Base
search: Base basename AbstractSet Broadcast broadcast broadcast! AbstractString AbstractDisplay set_zero_subnormals get_zero_subnormals

  No documentation found.

  No docstring found for module Base.

help?> Main
search: Main DomainError maxintfloat SegmentationFault maximum maximum! Matrix BitMatrix mapslices DenseMatrix timedwait StridedMatrix

  No documentation found.

  No docstring found for module Main.

help?> Core
search: Core isconcretetype searchsorted searchsortedlast searchsortedfirst code_lowered @code_lowered StackOverflowError code_warntype

  No documentation found.

  No docstring found for module Core.

help?> Main.Core
  No documentation found.

  No docstring found for module Core.

help?> Core.Main
  Main

  Main is the top-level module, and Julia starts with Main set as the current module. Variables defined at the prompt go in Main, and
  varinfo lists variables in Main.

  julia> @__MODULE__
  Main

help?> Main.Base
  No documentation found.

  No docstring found for module Base.

help?> Base
search: Base basename AbstractSet Broadcast broadcast broadcast! AbstractString AbstractDisplay set_zero_subnormals get_zero_subnormals

  No documentation found.

  No docstring found for module Base.
```

Reply:
I see. It seems to work if you document a module's "self binding", i.e. `Base.Base`, `Core.Core`, and `Core.Main.Main`.

Reply:
Cool, it works!